### ISSUE TITLE
unnecessary_wraps: ignore functions that return newtypes?

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6517]
````rust
type ThatsOk = Option<i32>;

fn new_ok() -> ThatsOk {
    Some(3)
}

fn main() {
    let _ = new_ok();
}
````

````
warning: this function's return value is unnecessarily wrapped by `Option`
 --> src/main.rs:3:1
  |
3 | / fn new_ok() -> ThatsOk {
4 | |     Some(3)
5 | | }
  | |_^
  |
  = note: `#[warn(clippy::unnecessary_wraps)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_wraps
help: remove `Option` from the return type...
  |
3 | fn new_ok() -> i32 {
  |                ^^^
help: ...and change the returning expressions
  |
4 |     3
  |
````

IMO  the lint should not trigger here because we use a newtype here and making code work without the newtype may take quite some effort which goes beyond removing a match or an `.unwrap()` in real world examples.

Thoughts?

**Comment by mikerite** [comment]
You're talking about aliases and not newtypes. This lint won't trigger on newtypes.

I'm against this proposal for two reasons.

1.The type alias doesn't actually make it more work to fix. You need to replace `ThatsOk` with `u32` whereas without the type alias you would have to replace `Option<u32>` with `u32`. It's the same amount of work.

2. "too much work" is really subjective. Fixing these little warnings might be too much work for me but not for you. I think we should just lint on all valid case and let the user of decide this for himself and `allow` or `deny` as appropriate.
